pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -56,6 +61,32 @@ func (r *RedisReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{}, err
 	}
 
+	// We want to reconcile redis-conf ConfigMap before Deployment
+	// to avoid restart redis pods twice in case of User change ConfigMap.
+	// Annotation GenerationID was added to Pod Template to support Upgrade scenario.
+	// this GenerationID is taken from ConfigMap resourceVersion.
+	// If User change Config Map, Operator will revert it to original one,
+	// but resourceVersion could be changed twice (after user change and after operator).
+	// To avoid this scenario by placing CM reconciliation before Deployment
+	if r.ConfigMap != nil {
+		redisConfigMap := r.ConfigMap(redis)
+		isInternalRedis, err := r.isInternalRedis()
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if isInternalRedis {
+			redis_configuration := r.ConfigMap(redis).Data["redis.conf"]
+			redisConfigMap.Data["redis.conf"] = redis_configuration + "\n" + "rename-command REPLICAOF \"\"" + "\n" + "rename-command SLAVEOF \"\"" + "\n"
+		}
+		err = r.ReconcileConfigMap(redisConfigMap, r.redisConfigCmMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if isInternalRedis {
+			r.RollOutRedisPods(context.TODO())
```

Comment:
err is not handled

Reply:
Thank you for comment. RollOutRedisPods is removed
Done